src/app.rs

```diff
@@ -3,47 +3,63 @@ use crossterm::event::{KeyCode, KeyEvent};
 use nalgebra::Point2;
 use rand::Rng;
 
-use crate::algorithms::graham_scan::GrahamScan;
+use crate::algorithms::{algorithm::AlgorithmWrapper, graham_scan::GrahamScan};
 
 pub enum InputMode {
     Normal,
     Editing,
 }
 
-pub struct TabsState<'a> {
-    pub titles: Vec<&'a str>,
+pub struct TabsState {
+    pub tabs: Vec<Tab>,
     pub index: usize,
 }
 
-impl<'a> TabsState<'a> {
-    pub fn new(titles: Vec<&'a str>) -> TabsState {
-        TabsState { titles, index: 0 }
+pub struct Tab {
+    pub algorithm: AlgorithmWrapper,
+    pub step: usize,
+    pub max_steps: Option<usize>,
+    pub point_amount: Option<usize>,
```

Comment:
This is alright, as long as we operate with algorithms which only use a point set for initialization.